client/components/data/query-themes/index.jsx

```diff
@@ -0,0 +1,54 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import shallowEqual from 'react-pure-render/shallowEqual';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { requestThemes } from 'state/themes/actions';
+import { isRequestingThemesForQuery } from 'state/themes/selectors';
+
+class QueryThemes extends Component {
+	static propTypes = {
+		siteId: PropTypes.oneOfType( [
+			PropTypes.number,
+			PropTypes.oneOf( [ 'wpcom' ] )
+		] ).isRequired,
+		query: PropTypes.object,
+		// Connected props
+		isRequesting: PropTypes.bool.isRequired,
+		requestThemes: PropTypes.func.isRequired,
+	}
```

Comment:
Heh, this caught me kinda off-guard since I'm not used to put any commas or anything after class methods. As for (static) class properties, turns out there's been a lengthy discussion on babel to first [implement an error](https://github.com/babel/babel/pull/3225) for a missing semicolon after a static class property, and then [to revert it](https://github.com/babel/babel/pull/3332). 

So for consistency with class methods, I'm inclined to not put a semicolon here...